ksml-integration-tests/src/test/java/io/axual/ksml/integration/BinaryDataFormatIT.java

```diff
@@ -0,0 +1,239 @@
+
+        // Prepare test environment with Binary-specific files
+        String resourcePath = "/docs-examples/beginner-tutorial/different-data-formats/binary";
+        String[] binaryFiles = {"ksml-runner.yaml", "binary-producer.yaml", "binary-processor.yaml"};
+
+        Path configPath = KSMLRunnerTestUtil.prepareTestEnvironment(
+                tempDir,
+                resourcePath,
+                binaryFiles,
+                kafka.getBootstrapServers()
+        );
+
+        log.info("Using KSMLRunner directly with config: {}", configPath);
+
+        // Start KSML using KSMLRunner main method
+        ksmlRunner = new KSMLRunnerWrapper(configPath);
+        ksmlRunner.start();
+
+        // Wait for KSML to be ready
+        waitForKSMLReady();
+    }
+
+    @Test
+    @Timeout(90) // 1.5 minutes should be enough for binary processing
+    void testKSMLBinaryProcessing() throws Exception {
+        // Wait for first binary data to be generated and processed
+        log.info("Waiting for KSML to generate and process binary data...");
+        waitForBinaryDataGeneration();
+
+        // Verify KSML is still running
+        assertThat(ksmlRunner.isRunning()).isTrue().as("KSMLRunner should still be running");
+
+        // Create consumer properties
+        Properties consumerProps = new Properties();
+        consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka.getBootstrapServers());
+        consumerProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        consumerProps.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        consumerProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class.getName());
+
+        // Check ksml_sensordata_binary topic (producer output - binary data)
+        consumerProps.put(ConsumerConfig.GROUP_ID_CONFIG, "test-consumer-binary");
+        Map<String, byte[]> originalMessages = new HashMap<>();
+        try (KafkaConsumer<String, byte[]> consumer = new KafkaConsumer<>(consumerProps)) {
+            consumer.subscribe(Collections.singletonList("ksml_sensordata_binary"));
+            ConsumerRecords<String, byte[]> records = KSMLRunnerTestUtil.pollWithRetry(consumer, Duration.ofSeconds(10));
+
+            assertThat(records).isNotEmpty().as("Should have generated binary data in ksml_sensordata_binary topic");
+            log.info("Found {} binary messages", records.count());
+
+            // Validate binary messages and store for comparison
+            records.forEach(record -> {
+                log.info("Binary message: key={}, bytes={}", record.key(), Arrays.toString(record.value()));
+                assertThat(record.key()).startsWith("msg").as("Message key should start with 'msg'");
+
+                // Validate binary structure - should be 7 bytes
+                byte[] binaryValue = record.value();
+                assertThat(binaryValue).hasSize(7).as("Binary message should have 7 bytes");
```

Comment:
using assertJ can actually compare the full byte array to an expected array.
Then this would become a single assertion, instead of multiple.